UKHomeOffice/cerberus-service
# COP-6747 Refactor target status

## Description
- Refactored how we determine target status and how we display/hide elements based on the target status
- Updated test fixtures to remove unneeded information related to displaying tasks/status
- Updated tests to ensure testing the various statuses

## To Test

- open a task from the `Issued` tab
- > you should have no `claim` or `unclaim` button
- > you should not see an `assigned to` message
- > you should not have the action buttons (Issue target, Assessment complete, Dismiss)
- > you should not have a notes form
- open a task from the 'Completed` tab
- > you should have no `claim` or `unclaim` button
- > you should not see an `assigned to` message
- > you should not have the action buttons (Issue target, Assessment complete, Dismiss)
- > you should not have a notes form
- open a task from the `in progress` tab that is not assigned to you
- > you should have no `claim` or `unclaim` button
- > you should not see the text 'Assigned to <user>'
- > you should not have the action buttons (Issue target, Assessment complete, Dismiss)
- > you should not have a notes form
- open a task from the `new` tab
- > you should have a `claim` button but no other buttons
- > you should see the text 'Unassigned'
- click `claim`
- > you should now see the text 'Assigned to you'
- > you should have an 'unclaim' button
- > you should have the action buttons
- > you should have a notes form

## Developer Checklist

\* Required
